openedx/features/course_duration_limits/models.py

```diff
@@ -48,6 +47,36 @@ class CourseDurationLimitConfig(StackedConfigurationModel):
         )
     )
 
+    @classmethod
+    def has_full_access_role_in_masquerade(cls, user, course_key, course_masquerade):
+        """
+        When masquerading as a user, the course duration limits will never trigger the course to expire, redirecting the user.
+        However, the user masquerade roles are still used to determine whether the course duration limits banner will display.
+        Enable the course duration limits banner for the enrollment if a masquerading user has any of the following roles
+        Staff, Instructor, Beta Tester, Forum Community TA, Forum Group Moderator, Forum Moderator, Forum Administrator
+        """
+        masquerade_role = get_masquerade_role(user, course_key)
+        verified_mode_id = settings.COURSE_ENROLLMENT_MODES.get(CourseMode.VERIFIED, {}).get('id')
+        # Masquerading users can select the the role of a verified users without selecting a specific user
+        is_verified = (course_masquerade.user_partition_id == ENROLLMENT_TRACK_PARTITION_ID
+                       and course_masquerade.group_id == verified_mode_id)
+        # Masquerading users can select the role of staff without selecting a specific user
+        is_staff = masquerade_role == 'staff'
+        # Masquerading users can select other full access roles for which content type gating is disabled
+        is_full_access = (course_masquerade.user_partition_id == CONTENT_GATING_PARTITION_ID
+                          and course_masquerade.group_id == CONTENT_TYPE_GATE_GROUP_IDS['full_access'])
+        # When masquerading as a specific user, we can check that user's staff roles as we would with a normal user
+        is_staff_role = False
+        try:
+            user = User.objects.get(username=course_masquerade.user_name)
```

Comment:
At this point `user` should be the target user, not the staff user, so I don't think this lookup is necessary?